agent/agentserver/server_test.go

```diff
@@ -207,6 +211,72 @@ func TestHealthHandler(t *testing.T) {
 	}
 }
 
+func TestReadinessCheckHandler(t *testing.T) {
+	for _, tc := range []struct {
+		desc                  string
+		probeErr              error
+		buildIndexErr         error
+		trackerErr            error
+		expectedErrMsgPattern string
+	}{
+		{
+			desc:                  "success",
+			probeErr:              nil,
+			buildIndexErr:         nil,
+			trackerErr:            nil,
+			expectedErrMsgPattern: "",
+		},
+		{
+			desc:                  "failure (probe fails)",
+			probeErr:              errors.New("test scheduler error"),
+			buildIndexErr:         nil,
+			trackerErr:            nil,
+			expectedErrMsgPattern: `GET http://127\.0\.0\.1:\d+/readiness 503: agent not ready: test scheduler error`,
+		},
+		{
+			desc:                  "failure (build index not ready)",
+			probeErr:              nil,
+			buildIndexErr:         errors.New("build index not ready"),
+			trackerErr:            nil,
+			expectedErrMsgPattern: `GET http://127\.0\.0\.1:\d+/readiness 503: agent not ready: build index not ready`,
+		},
+		{
+			desc:                  "failure (tracker not ready)",
+			probeErr:              nil,
+			buildIndexErr:         nil,
+			trackerErr:            errors.New("tracker not ready"),
+			expectedErrMsgPattern: `GET http://127\.0\.0\.1:\d+/readiness 503: agent not ready: tracker not ready`,
+		},
+		{
+			desc:                  "failure (all conditions fail)",
+			probeErr:              errors.New("test scheduler error"),
+			buildIndexErr:         errors.New("build index not ready"),
+			trackerErr:            errors.New("tracker not ready"),
+			expectedErrMsgPattern: `GET http://127\.0\.0\.1:\d+/readiness 503: agent not ready: test scheduler error\nbuild index not ready\ntracker not ready`,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			require := require.New(t)
+
+			mocks, cleanup := newServerMocks(t)
+			defer cleanup()
+
+			mocks.sched.EXPECT().Probe().Return(tc.probeErr)
+			mocks.tags.EXPECT().CheckReadiness().Return(tc.buildIndexErr)
+			mocks.ac.EXPECT().CheckReadiness().Return(tc.trackerErr)
+
+			addr := mocks.startServer()
+			_, err := httputil.Get(fmt.Sprintf("http://%s/readiness", addr))
+			if tc.expectedErrMsgPattern == "" {
+				require.Nil(err)
+			} else {
+				r, _ := regexp.Compile(tc.expectedErrMsgPattern)
```

Comment:
instead of a regex you could assert the full error string just do a strings.ReplaceAll on a special pattern for the address

```
require.EqualError(t, err, strings.ReplaceAll(tt.wantErr, "{address}", addr))
```

and the wantErr value would be

```
"GET http://{address}/readiness 503: agent not ready: test scheduler error"
```

Reply:
Much better this way, done.